### PR TITLE
Reset state machine when apply snapshot from leader

### LINKED ISSUE #99: `numChildren` become a negative number
### RaftKeeper version
raftkeeper:1.0.4
### Description
stat  /ckpub214.olap.jd.com/tables/sr_data_ofl_mix/app_sr_rec_other_expo_detail/05/replicas/00
cZxid = 0xcbef515
ctime = Thu Oct 19 19:05:24 CST 2023
mZxid = 0xcbef515
mtime = Thu Oct 19 19:05:24 CST 2023
pZxid = 0x27c63e9fb
cversion = 287
dataVersion = 0
aclVersion = 0
ephemeralOwner = 0x0
dataLength = 0
numChildren = -1

### PR DESCRIPTION
**Pull request by JackyWoo**
### Which issues of this PR fixes:

This PR try to fix #99 

### Change log:

- Reset state machine when apply snapshot from leader

/cc @lzydmxy

### CODE REVIEW

File: src/Service/NuRaftStateMachine.cpp [hunk 1]
```diff
@@ -753,9 +753,13 @@ bool NuRaftStateMachine::existSnapshotObject(snapshot & s, ulong obj_id)
 
 bool NuRaftStateMachine::apply_snapshot(snapshot & s)
 {
-    /// TODO: double buffer load or multi thread load
+    /// If the invoker is from NuRaft, we should reset the state machine
+    LOG_INFO(log, "reset state machine.");
+    reset();
```
Comment (lzydmxy) [inline src/Service/NuRaftStateMachine.cpp:758]: Should we get `snapshot_mutex` before reset?

Comment (JackyWoo) [reply src/Service/NuRaftStateMachine.cpp:760]: Fixed thanks

**Comment by JackyWoo** [comment]
@marising setted

**Comment by JackyWoo** [comment]
Please ignore the tsan error, it will be fixed in another PR.

**Pull request merged**